### PR TITLE
chore(profiling): remove attrs from ddtrace/profiling

### PR DESCRIPTION
**Pull request by taegyunkim**
## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/internal/telemetry/writer.py [hunk 1]
```diff
@@ -234,7 +234,7 @@ class TelemetryWriter(PeriodicService):
 
     def __init__(self, is_periodic=True, agentless=None):
         # type: (bool, Optional[bool]) -> None
-        super(TelemetryWriter, self).__init__(interval=min(config._telemetry_heartbeat_interval, 10))
+        super(TelemetryWriter, self).__init__(min(config._telemetry_heartbeat_interval, 10))
```
Comment (P403n1x87) [inline ddtrace/internal/telemetry/writer.py:237]:
```suggestion
        super().__init__(min(config._telemetry_heartbeat_interval, 10))
```

File: ddtrace/internal/writer/writer.py [hunk 2]
```diff
@@ -161,6 +161,8 @@ def __init__(
         report_metrics=True,  # type: bool
     ):
         # type: (...) -> None
+        super(HTTPWriter, self).__init__()
+        TraceWriter.__init__(self)
```
Comment (P403n1x87) [inline ddtrace/internal/writer/writer.py:165]: 🤔  we shouldn't be needing to call all the `__init__`s manually if every other `__init__` in the mro calls `super` I think

File: ddtrace/profiling/_threading.pyx [hunk 3]
```diff
@@ -103,16 +103,16 @@ else:
...
-@attr.s(slots=True, eq=False)
 class _ThreadLink(_thread_link_base):
     """Link a thread with an object.
 
     Object is removed when the thread disappears.
     """
+    __slots__ = ("_thread_id_to_object",)
+
+    def __init__(self):
+        self._thread_id_to_object: typing.Dict[int, _weakref_type] = {}
```
Comment (P403n1x87) [inline ddtrace/profiling/_threading.pyx:114]: We should add a call to `super` to ensure inheritance works as expected perhaps

**Comment by taegyunkim** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/DataDog/dd-trace-py/9796?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#9796** <a href="https://app.graphite.dev/github/pr/DataDog/dd-trace-py/9796?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#9821** <a href="https://app.graphite.dev/github/pr/DataDog/dd-trace-py/9821?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @taegyunkim and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by brettlangdon — APPROVED** [review]
`ddtrace/internal/` files lgtm

**Review by sanchda — APPROVED** [review]
LGTM!  Thank you!!!

**Pull request merged**